dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,163 @@
+.AddOptionalArg<int>("in_anchors", R"code(2D TensorList of type int64
+
+Absolute values of LU corner of the selection for each iteration.
+Zeros are used if this is omitted.)code", nullptr, true)
+.AddOptionalArg<int>("shapes", R"code(2D TensorList of type int64
+
+Absolute values of size of the selection for each iteration.
+Input size is used if this is omitted.)code", nullptr, true)
+.AddOptionalArg<int>("out_anchors", R"code(2D TensorList of type int64
+
+Absolute values of LU corner of the paste for each iteration.
+Zeros are used if omitted.)code", nullptr, true)
+.AddArg("output_size",
+R"code(Output size.)code", DALI_INT_VEC, true)
+.AddOptionalArg("dtype",
+R"code(Output data type. If not set, the input type is used.)code", DALI_NO_TYPE)
+.NumOutput(1);
+
+DALI_REGISTER_OPERATOR(MultiPaste, MultiPasteCpu, CPU)
+
+bool MultiPasteCpu::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                      const workspace_t<CPUBackend> &ws) {
+  AcquireArguments(spec_, ws);
+
+  const auto &images = ws.template InputRef<CPUBackend>(0);
+  const auto &output = ws.template OutputRef<CPUBackend>(0);
+  output_desc.resize(1);
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = TheKernel<OutputType, InputType>;
+            kernel_manager_.Initialize<Kernel>();
+            TensorListShape<> sh = images.shape();
+            TensorListShape<> shapes(sh.num_samples(), sh.sample_dim());
+            for (int i = 0; i < sh.num_samples(); i++) {
+              const TensorShape<> &out_sh =
+                dali::TensorShape<>(output_size_[i].data[0], output_size_[i].data[1], sh[i][2]);
+              shapes.set_tensor_shape(i, out_sh);
+            }
+
+            TypeInfo type;
+            type.SetType<OutputType>(output_type_);
+            output_desc[0] = {shapes, type};
+          }
+      ), DALI_FAIL(make_string("Unsupported output type: ", output_type_)))  // NOLINT
+  ), DALI_FAIL(make_string("Unsupported input type: ", images.type().id())))  // NOLINT
+  return true;
+}
+
+
+
+
+void MultiPasteCpu::RunImpl(workspace_t<CPUBackend> &ws) {
+  const auto &images = ws.template InputRef<CPUBackend>(0);
+  auto &output = ws.template OutputRef<CPUBackend>(0);
+
+  output.SetLayout(images.GetLayout());
+  auto out_shape = output.shape();
+
+  auto& tp = ws.GetThreadPool();
+
+  auto batch_size = output.shape().num_samples();
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = TheKernel<OutputType, InputType>;
+            for (int i = 0; i < batch_size; i++) {
+              auto paste_count = in_idx_[i].shape[0];
+
+              auto to_zero = view<OutputType, 3>(output[i]);
```

Comment:
It's better to create a view of the whole batch outside of the loop, and then access each sample:
```
auto out_view = view<OutputType, 3>(output);
for (int i=0; ...) {
  out_view[i]
  }
  ```